xtuner/v1/train/trainer.py

```diff
@@ -327,16 +321,26 @@ def __init__(
 
         self._resolve_config_conflicts(self.tokenizer, model_cfg, dataloader_cfg)
 
-        self._dataloader = self.build_dataloader(
-            dataset_config=dataset_cfg,
-            dataloader_config=dataloader_cfg,
-            dp_mesh=self.data_mesh["dp"],
+        if dataset_cfg is not None:
```

Comment:
```suggestion
        if dataset_cfg is not None:  # TODO: Removed in version 1.1.0
```

And raise a warning here